boilerplate/fullstack/remix/app/routes/_index.tsx

```diff
@@ -0,0 +1,194 @@
+        console.log("does the user have invalid claims?", hasInvalidClaims);
+        console.log("does the user have an access token??", hasToken);
+        if (session) {
+            console.log("there is an active session");
+        }
+        if (!session) {
+            console.log("session does not exist or has expired");
+        }
+
+        const res: SessionForRemixProps = {
+            session: {
+                userId: session?.getUserId(),
+                sessionHandle: session?.getHandle(),
+                accessTokenPayload: session?.getAccessTokenPayload(),
+            },
+            hasInvalidClaims,
+            hasToken,
+        };
+
+        console.log("\n\n", res, "\n\n");
+
+        if (nextResponse) {
+            console.log("nextResponse is:", nextResponse);
+            return {
+                // session,
+                session: res,
+                hasInvalidClaims,
+                hasToken,
+                nextResponse,
+            };
+        } else {
+            console.log("nextResponse is null");
+            return {
+                session: res,
+                hasInvalidClaims,
+                hasToken,
+                nextResponse: null,
+            };
+        }
+    } catch (error) {
+        console.error("Error retrieving session:", error);
+        throw error;
+    }
+}
+
+export default function Home() {
+    const loaderData = useLoaderData<{
+        session: SessionForRemixProps | undefined;
+        hasInvalidClaims: boolean;
+        hasToken: boolean;
+        nextResponse: Response | null;
+    }>();
+
+    async function logoutClicked() {
+        await SessionReact.signOut();
+        SuperTokens.redirectToAuth();
+    }
+
+    if (loaderData.nextResponse) {
+        return (
+            <div>
+                Something went wrong while trying to get the session. Error -{loaderData.nextResponse.status}{" "}
+                {loaderData.nextResponse.statusText}
+            </div>
+        );
+    }
+
+    if (!loaderData.session) {
+        if (!loaderData.hasToken) {
+            console.log("Redirecting to /auth");
+            return redirect("/auth");
+        }
+        if (loaderData.hasInvalidClaims) {
+            console.log("Session has invalid claims");
+            return <SessionAuthForRemix />;
+        } else {
+            console.log("Trying to refresh session");
+            return <TryRefreshComponent />;
+        }
+    }
+    if (loaderData.session.session) {
+        const sessionData: SessionDataForUI = {
+            note: "Retrieve authenticated user-specific data from your application post-verification through the use of the verifySession middleware.",
+            userId: loaderData.session.session?.userId || "",
+            sessionHandle: loaderData.session.session?.accessTokenPayload.sessionHandle,
+            accessTokenPayload: loaderData.session.session?.accessTokenPayload,
+        };
```

Comment:
no need to do `|| ""` cause session should always be here. So ideally, you just do loaderData.session.userId. Of ideally, just pass the session container itself and do loaderData.session.getUserId()

Reply:
✔️ - but can't pass the entire session container as it gets converted to playing JSON client-side (causing TS to complain)